chore: upgrade next, add linting

tellaho opened the pull request:
- Upgrades `next` v13 → v15, kept at `react` v18
- turns on [Turbopack](https://nextjs.org/docs/app/api-reference/turbopack)
- removes `babel`, ups `react-email` dependency
- Adds linting

tellaho committed:
chore: update nextjs codemod + react + deps

`npx @next/codemod@canary upgrade latest`

tellaho committed:
chore: cleanup

tellaho committed:
fix: icon was not displaying

stopped working after migration?

tellaho committed:
Merge branch 'main' into tho-next-v15

tellaho committed:
fix: more fixes

tellaho committed:
feat: add linting

tellaho committed:
chore: run `pnpm run lint:fix`

tellaho committed:
chore: update README (`pnpm`, new linting)

tellaho committed:
chore: more lint

tellaho committed:
Merge branch 'main' into tho-next-v15

tellaho committed:
fix: more defensive

tellaho committed:
feat: improve null safety and error handling across components

- Fix router navigation issue in MemberEdit
- Add null safety to FilterPicker and data handling
- Improve Firebase helper validation and error handling
- Update type imports and add null filtering
- Enhance data handling robustness

tellaho committed:
feat: enhance member update logic robustness

- Improve handling of single vs. array values for regions/experience
- Add better type checking for field values
- Enhance reference ID extraction logic
- Make member update process more robust against edge cases

tellaho committed:
feat: better linting

tellaho committed:
chore: update readme

tellaho commented:
@eparubrub re-up for a review when you have a chance!

eparubrub commented on pages/admin/directory.tsx:
```diff
@@ -84,18 +74,33 @@ export default function DirectoryPage(props) {
+            if (bIndex === -1) return -1; // Put undefined names at the end
+
+            return aIndex - bIndex;
+          }), // sort experience filter explicitly
```
Since we're also doing this similar experience filtering in `components/filters/FilterPicker.tsx` and `pages/edit/member.tsx`, we should make a helper function for this

tellaho replied:
Thanks for catching that! I've created two helper functions to eliminate the duplicate sorting logic:
- `sortByOrder` for complete arrays (used in member.tsx and admin/directory.tsx)
- `compareByOrder` for sort callbacks (used in FilterPicker.tsx)

Both handle the same core logic but serve different use cases.

eparubrub commented on lib/firebase-helpers/members.ts:
```diff
@@ -291,21 +313,7 @@ export const updateMember = async (
-    regions,
-    ...droppedMemberData
-  } = memberData;
+  const { emailAbbr, companySize, ...droppedMemberData } = memberData;
```
I think we have to revert this specific change

If these fields aren't dropped then unnecessary field data is saved to the database. Look at the fields that get  saved to the database after updating a member

tellaho replied:
great catch thank you!

eparubrub reviewed (changes requested):
Just one request to revert the change on member edit. The other one is just a suggestion for DRY principle but non-blocking

Other than that I was able to run locally and run the linting commands

tellaho committed:
chore: added sort utils

tellaho committed:
revert: extracts fields that should be dropped from memberData

tellaho committed:
chore: omitting fields with better type safety

tellaho commented:
### better destructuring and added type safety
Thanks for the call-out. The simplified version was accidentally including unnecessary fields like `focus`, `industry`, and `id`. I've reverted to the explicit field removal approach (using `delete` like we do in the `addMember` function above) and added compile-time type safety using `Omit<MemberPublic, FieldsToExclude>` to catch any future mistakes at build time.

### dup sorting functions
I've created two helper functions to eliminate the duplicate sorting logic:
- `sortByOrder` for complete arrays (used in member.tsx and admin/directory.tsx)
- `compareByOrder` for sort callbacks (used in FilterPicker.tsx)
Both handle the same core logic but serve different use cases. This should make the code more maintainable across the codebase.
